Objects/typeobject.c

```diff
@@ -9358,14 +9358,15 @@ super_repr(PyObject *self)
 }
 
 static PyObject *
-super_getattro(PyObject *self, PyObject *name)
+do_super_lookup(superobject *su, PyTypeObject *su_type, PyObject *su_obj,
+                PyTypeObject *su_obj_type, PyObject *name, int *meth_found)
```

Comment:
It looks like `meth_found` is serving double duty as a flag, (meth_found != NULL means this a method lookup), and as a return value.
I think those should be separate arguments.

Reply:
Alternatively, replace `meth_found` with the `self` output argument. 
If `self` is not wanted (attribute, not method, lookup) then passing `NULL` would be OK, as the semantics are clearer.
```C
do_super_lookup(PyTypeObject *su_class, PyObject *su_obj,
                PyTypeObject *su_obj_class, PyObject *name, PyObject **self)
```

Reply:
> Is su_obj_type always Py_TYPE(su_obj)?

Unfortunately not; sometimes `su_obj_type == su_obj` (in the case where `su_obj` is already a class); see `supercheck`, which decides what `su_obj_type` should be, based on `su_type` and `su_obj`.

Reply:
I'll try replacing `int *meth_found` with `PyObject **self` and see how that looks. It's a little odd to have an output argument whose value (if set) will always be one of the input arguments (`su_obj` in this case), but I agree that the semantics are a bit clearer than `meth_found`.

Reply:
After playing with the `**self` option, I don't like it because the ref-counting semantics are hard to manage in a good way. The arguments to `do_super_lookup` are borrowed; if it sets the `**self` output argument, it really should incref it and return a "new" ref, but that just introduces a totally unnecessary incref/decref pair. So I'd rather stick with a simple flag argument.

I really don't think that clarity requires splitting `meth_found` into two arguments. The semantics of "we'll pass this output argument pointer in the case we are looking for a method, otherwise we'll set it NULL" aren't _that_ strange or confusing. Splitting it into two arguments needlessly introduces the possibility of nonsensical combinations of arguments (it doesn't make sense to pass an output arg pointer when we don't want that output arg, and it doesn't make sense to set it NULL if we do want it), which is bad API design. I think all that's needed here for sufficient clarity is a rename of `meth_found` and a comment explaining the semantics.